modules/gapi/src/backends/fluid/gfluidcore_simd_sse41.hpp

```diff
@@ -554,7 +554,162 @@ CV_ALWAYS_INLINE void calcRowLinear_8UC_Impl_<3>(uint8_t* dst[],
         }
     }
 }
-} // namespace sse42
+
+CV_ALWAYS_INLINE void v_gather_pairs(const float src[], const int* mapsx,
+                                     v_float32x4& low, v_float32x4& high)
+{
+    __m128 l = _mm_setzero_ps();
+    l = _mm_loadl_pi(l, (const __m64*)&src[mapsx[0]]);  // pair of floats
+    l = _mm_loadh_pi(l, (const __m64*)&src[mapsx[1]]);
+    low.val = l;
+
+    __m128 h = _mm_setzero_ps();
+    h = _mm_loadl_pi(h, (const __m64*)&src[mapsx[2]]);
+    h = _mm_loadh_pi(h, (const __m64*)&src[mapsx[3]]);
+    high.val = h;
+}
+
+CV_ALWAYS_INLINE void v_deinterleave(const v_float32x4& low, const v_float32x4& high,
+                                     v_float32x4& even,      v_float32x4& odd)
+{
+    __m128 tmp0 = _mm_unpacklo_ps(low.val, high.val);
+    __m128 tmp1 = _mm_unpackhi_ps(low.val, high.val);
+    even.val = _mm_unpacklo_ps(tmp0, tmp1);
+    odd .val = _mm_unpackhi_ps(tmp0, tmp1);
+}
+
+// Resize (bi-linear, 32FC1)
+CV_ALWAYS_INLINE void calcRowLinear32FC1Impl(float *dst[],
+                                             const float *src0[],
+                                             const float *src1[],
+                                             const float  alpha[],
+                                             const int    mapsx[],
+                                             const float  beta[],
+                                             const Size& inSz,
+                                             const Size& outSz,
+                                             const int   lpi)
+{
+    bool xRatioEq1 = inSz.width == outSz.width;
+    bool yRatioEq1 = inSz.height == outSz.height;
+
+    constexpr int nlanes = v_float32x4::nlanes;
+
+    if (!xRatioEq1 && !yRatioEq1)
+    {
+        for (int line = 0; line < lpi; ++line) {
+            float beta0 = beta[line];
+            float beta1 = 1 - beta0;
+            v_float32x4 v_beta0 = v_setall_f32(beta0);
+            int x = 0;
+
+            v_float32x4 low1, high1, s00, s01;
+            v_float32x4 low2, high2, s10, s11;
+            for (; x <= outSz.width - nlanes; x += nlanes)
+            {
+                v_float32x4 alpha0 = v_load(&alpha[x]);
+                //  v_float32 alpha1 = 1.f - alpha0;
+
+                v_gather_pairs(src0[line], &mapsx[x], low1, high1);
+                v_deinterleave(low1, high1, s00, s01);
+
+                //  v_float32 res0 = s00*alpha0 + s01*alpha1;
+                v_float32x4 res0 = v_fma(s00 - s01, alpha0, s01);
+
+                v_gather_pairs(src1[line], &mapsx[x], low2, high2);
```

Comment:
If you do not rely on compiler optimization still  then don't you think to rearrange operations like  this?
```
gather_pairs(...low1, high1);
gather_pairs(...low2, high2)
v_deinterleave
v_deinterleave
...
v_fma
v_fma
```
it *may* exclude  cache pressure on mapsx